File: TodoListProject/src/main/java/com/project/taskmanagement/service/AnalyticsService.java
package com.project.taskmanagement.service;

import com.project.taskmanagement.dto.response.analytics.BurnupChartResponse;
import com.project.taskmanagement.dto.response.analytics.CumulativeFlowResponse;
import com.project.taskmanagement.dto.response.analytics.ProjectAnalyticsSummaryResponse;
import com.project.taskmanagement.dto.response.analytics.VelocityChartResponse;

import java.util.UUID;

public interface AnalyticsService {

    VelocityChartResponse getVelocity(UUID projectId);

    BurnupChartResponse getProjectBurnup(UUID projectId);

    BurnupChartResponse getSprintBurnup(UUID projectId, UUID sprintId);

    CumulativeFlowResponse getSprintCumulativeFlow(UUID projectId, UUID sprintId);

    ProjectAnalyticsSummaryResponse getProjectAnalyticsSummary(UUID projectId);
}

File: Fe/src/features/project/services/analytics.service.ts
import { apiRequest } from '../../../services/apiClient'
import { projectPath } from '../../../services/endpoints'
import type {
  ProjectAnalyticsSummaryResponse,
  VelocityChartResponse,
  BurnupChartResponse,
  CumulativeFlowResponse
} from '../models/analytics.model'

export const getProjectAnalyticsSummary = (projectId: string) =>
  apiRequest<ProjectAnalyticsSummaryResponse>(`${projectPath(projectId)}/analytics/summary`)

export const getProjectVelocity = (projectId: string) =>
  apiRequest<VelocityChartResponse>(`${projectPath(projectId)}/analytics/velocity`)

export const getProjectBurnup = (projectId: string) =>
  apiRequest<BurnupChartResponse>(`${projectPath(projectId)}/analytics/burnup`)

export const getSprintBurnup = (projectId: string, sprintId: string) =>
  apiRequest<BurnupChartResponse>(`${projectPath(projectId)}/sprints/${sprintId}/analytics/burnup`)

export const getSprintCumulativeFlow = (projectId: string, sprintId: string) =>
  apiRequest<CumulativeFlowResponse>(`${projectPath(projectId)}/sprints/${sprintId}/analytics/cumulative-flow`)
